### DISCUSSION TITLE
[QUESTION] How to implement object model object permissions like DRF?

### DISCUSSION THREAD

**Discussion by abhi1693** [NilCoalescing/djangochannelsrestframework#95] [Q&A]

**Comment by hishnash** [comment]
Currently we do not have an easy solution to support drop in object permissions like DRF however if you create a mixin you can do the object checks each instance. 

```python
class CheckObjectPermissions:
    def get_object(self, **kwargs) -> Model:
        obj = super().get_object(**kwargs)
        # do your object permission checks here throw regular permission error if needed.
        return obj

```

For list actions you will need to overhead the `get_serializer` method to add extra filtering.
